avro-converter/src/main/java/io/confluent/connect/avro/AvroData.java

```diff
@@ -619,82 +676,121 @@ private static Object maybeWrapSchemaless(Schema schema, Object value, String ty
-    // struct converts to one of those (record), so for everything else that has a name we store
-    // the full name into a special property. For uniformity, we also duplicate this info into
-    // the same field in records as well even though it will also be available in the namespace()
-    // and name().
-    if (schema.name() != null) {
-      baseSchema.addProp(CONNECT_NAME_PROP, schema.name());
-
-      if(Decimal.LOGICAL_NAME.equalsIgnoreCase(schema.name())){
-        baseSchema.addProp(AVRO_LOGICAL_TYPE_PROP, AVRO_LOGICAL_DECIMAL);
-      } else if(Time.LOGICAL_NAME.equalsIgnoreCase(schema.name())) {
-        baseSchema.addProp(AVRO_LOGICAL_TYPE_PROP, AVRO_LOGICAL_TIME_MILLIS);
-      } else if(Timestamp.LOGICAL_NAME.equalsIgnoreCase(schema.name())) {
-        baseSchema.addProp(AVRO_LOGICAL_TYPE_PROP, AVRO_LOGICAL_TIMESTAMP_MILLIS);
-      } else if(Date.LOGICAL_NAME.equalsIgnoreCase(schema.name())) {
-        baseSchema.addProp(AVRO_LOGICAL_TYPE_PROP, AVRO_LOGICAL_DATE);
+        if (schema.name() != null) {
+          baseSchema.addProp(CONNECT_NAME_PROP, schema.name());
+        }
+        // Some Connect types need special annotations to preserve the types accurate due to
+        // limitations in Avro. These types get an extra annotation with their Connect type
+        if (connectType != null) {
+          baseSchema.addProp(CONNECT_TYPE_PROP, connectType);
+        }
+      }
+      
+      // Only Avro named types (record, enum, fixed) may contain namespace + name. Only Connect's
+      // struct converts to one of those (record), so for everything else that has a name we store
+      // the full name into a special property. For uniformity, we also duplicate this info into
+      // the same field in records as well even though it will also be available in the namespace()
+      // and name().
+      if (schema.name() != null) {
+        if(Decimal.LOGICAL_NAME.equalsIgnoreCase(schema.name())){
+          baseSchema.addProp(AVRO_LOGICAL_TYPE_PROP, AVRO_LOGICAL_DECIMAL);
+        } else if(Time.LOGICAL_NAME.equalsIgnoreCase(schema.name())) {
+          baseSchema.addProp(AVRO_LOGICAL_TYPE_PROP, AVRO_LOGICAL_TIME_MILLIS);
+        } else if(Timestamp.LOGICAL_NAME.equalsIgnoreCase(schema.name())) {
+          baseSchema.addProp(AVRO_LOGICAL_TYPE_PROP, AVRO_LOGICAL_TIMESTAMP_MILLIS);
+        } else if(Date.LOGICAL_NAME.equalsIgnoreCase(schema.name())) {
+          baseSchema.addProp(AVRO_LOGICAL_TYPE_PROP, AVRO_LOGICAL_DATE);
+        }
       }
-    }
 
-    // Some Connect types need special annotations to preserve the types accurate due to
-    // limitations in Avro. These types get an extra annotation with their Connect type
-    if (connectType != null) {
-      baseSchema.addProp(CONNECT_TYPE_PROP, connectType);
-    }
+      if (schema.parameters() != null){
+        for(Map.Entry<String, String> entry : schema.parameters().entrySet())
+        {
+          if (entry.getKey().startsWith(AVRO_PROP)){
```

Comment:
avro. namespace whilst isn't in spec, i believe by convention is used for avro language specific extensions. e.g. avro.<language>.extension, obviously the java.string one actually is written in spec, i saw though cannot recall their Avro JIRA's for additional ones. 

Saying this, we can make it specific to just avro.java.string if we want to be spec/doc only features, just we will need to update as we find more by happenstance where not documented but used.